### PR TITLE
thrashosds: send a random osd a signal.SIGHUP

### PR DESCRIPTION
**Pull request by andrewschoen**
See: http://tracker.ceph.com/issues/12476

### CODE REVIEW

File: tasks/ceph_manager.py [hunk 1]
```diff
@@ -135,6 +137,8 @@ def tmp(x):
             manager.raw_cluster_cmd('--', 'mon', 'tell', '*', 'injectargs',
                                     '--mon-osd-down-out-interval 0')
         self.thread = gevent.spawn(self.do_thrash)
+        if self.hangup:
```
Comment (zmc) [inline tasks/ceph_manager.py]: ``self.hangup_delay``

File: tasks/ceph_manager.py [hunk 2]
```diff
@@ -425,6 +429,8 @@ def do_join(self):
         """
         self.stopping = True
         self.thread.get()
+        if self.hangup:
```
Comment (zmc) [inline tasks/ceph_manager.py]: ``self.hangup_delay``

**Comment by dmick** [comment]
I'd recommend "hup" rather than "hangup" for familiarity.  looking deeper

File: tasks/ceph_manager.py (hunk 1 above)
Comment (andrewschoen) [inline tasks/ceph_manager.py]: ah, thanks!

File: tasks/ceph_manager.py [hunk 3]
```diff
@@ -112,6 +113,7 @@ def __init__(self, manager, config, logger=None):
         self.clean_wait = self.config.get('clean_wait', 0)
         self.minin = self.config.get("min_in", 3)
         self.chance_move_pg = self.config.get('chance_move_pg', 1.0)
+        self.hangup_delay = self.config.get('hangup_delay')
```
Comment (zmc) [inline tasks/ceph_manager.py]: might want to ``float()`` this

**Comment by andrewschoen** [comment]
I'm fine with ``hup_delay`` over ``hangup_delay``. @zmc, thoughts?

File: tasks/ceph_manager.py (hunk 3 above)
Comment (dmick) [inline tasks/ceph_manager.py]: default to 0 explicitly?

Comment (andrewschoen) [inline tasks/ceph_manager.py]: Good call, I won't here but when I use it in time.sleep I will.  I want a return value of None here to be ok.

Comment (zmc) [inline tasks/ceph_manager.py]: I don't see an issue with defaulting to ``None``

File: tasks/thrashosds.py [hunk 4]
```diff
@@ -112,6 +115,8 @@ def task(ctx, config):
         config = {}
     assert isinstance(config, dict), \
         'thrashosds task only accepts a dict for configuration'
+    # add default value for hangup_delay
+    config['hangup_delay'] = config.get('hangup_delay', 0.1)
```
Comment (dmick) [inline tasks/thrashosds.py]: hm, now I see this default.  Do we always want this behavior for thrash?

Comment (zmc) [inline tasks/thrashosds.py]: that seems to be what's being asked for

Comment (liewegas) [inline tasks/thrashosds.py]: yes

**Comment by andrewschoen** [comment]
Alright, I've changed everything from 'hangup' to 'sighup'.

**Comment by dmick** [comment]
:thumbsup:

**Comment by andrewschoen** [comment]
This looks to be working as expected.

http://pulpito.ceph.com/sage-2015-07-28_19:05:15-rados-wip-12465-distro-basic-multi/

**Pull request merged**